zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ServiceTaskTest.java

```diff
@@ -0,0 +1,80 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tngtech.archunit.thirdparty.com.google.common.collect.Lists;
+import io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnJobBehavior.LinkedResourceProps;
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
+import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeBindingType;
+import io.camunda.zeebe.protocol.record.Assertions;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.protocol.record.value.JobRecordValue;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.Map;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class ServiceTaskTest {
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+  private static final String PROCESS_ID = "process";
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldCompleteJobWithLinkedResources() throws JsonProcessingException {
+    final BpmnModelInstance modelInstance =
+        Bpmn.createExecutableProcess("process")
+            .startEvent()
+            .serviceTask(
+                "my_linked_resource",
+                t ->
+                    t.zeebeLinkedResources(
+                            l ->
+                                l.resourceId("id")
+                                    .resourceType("RPA")
+                                    .bindingType(ZeebeBindingType.deployment)
+                                    .versionTag("1v")
+                                    .linkName("my_link"))
+                        .zeebeJobType("type"))
+            .endEvent()
+            .done();
+
+    ENGINE.deployment().withXmlResource(modelInstance).deploy();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    ENGINE.job().ofInstance(processInstanceKey).withType("type").complete();
+
+    // then
+    final Record<JobRecordValue> jobCreated =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+    final LinkedResourceProps resourceProps = new LinkedResourceProps();
+    resourceProps.setResourceType("RPA");
+    resourceProps.setLinkName("my_link");
+    // TODO: key lookup will be implemented in the next PR
+    resourceProps.setResourceKey("");
+
+    Assertions.assertThat(jobCreated.getValue())
+        .hasCustomHeaders(
+            Map.of(
+                "linkedResources", MAPPER.writeValueAsString(Lists.newArrayList(resourceProps))));
```

Comment:
🔧 We can just use the standard library's `List.of` instead of importing the third party `com.tngtech.archunit.thirdparty.com.google.common.collect.Lists`

```suggestion
            Map.of("linkedResources", MAPPER.writeValueAsString(List.of(resourceProps))));
```

Reply:
That is an accidental import :(